spaceship/lib/spaceship/two_step_client.rb

```diff
@@ -36,12 +36,53 @@ def handle_two_step(response)
         device_id = result.match(/.*\t.*\t\((.*)\)/)[1]
         select_device(r, device_id)
       elsif r.body.kind_of?(Hash) && r.body["phoneNumberVerification"].kind_of?(Hash)
-        raise "spaceship currently doesn't support the push based 2 step verification, please switch to SMS based 2 factor auth in the mean-time"
+        puts "Two Fator Authentication for account '#{self.user}' is enabled"
+        handle_two_fator(r)
       else
         raise "Invalid 2 step response #{r.body}"
       end
     end
 
+    def handle_two_fator(response)
+      security_code = response.body["phoneNumberVerification"]["securityCode"]
+      # {"length"=>6,
+      #  "tooManyCodesSent"=>false,
+      #  "tooManyCodesValidated"=>false,
+      #  "securityCodeLocked"=>false}
+      code_length = security_code["length"]
+      puts "Successfully requested notification"
+      code = ask("Please enter the #{code_length} digit code: ")
+      puts "Requesting session..."
+
+      # Send securityCode back to server to get a valid session
+      r = request(:post) do |req|
+        req.url "https://idmsa.apple.com/appleauth/auth/verify/trusteddevice/securitycode"
+        req.headers["Accept"] = "application/json"
+        req.headers['Content-Type'] = 'application/json'
+        req.headers["scnt"] = @scnt
+        req.headers["X-Apple-Id-Session-Id"] = @x_apple_id_session_id
+        req.body = { "securityCode" => { "code" => code.to_s } }.to_json
+      end
+
+      # we use `Spaceship::TunesClient.new.handle_itc_response`
```

Comment:
Those comments are relevant, it's little different with two-step.
Two-factor use `securityCode`, two-step use `code`